dashboard/src/components/Config/AppRepoList/AppRepoForm.test.tsx

```diff
@@ -699,13 +780,16 @@ describe("when the repository info is already populated", () => {
           filterRule: { jq: ".name | test($var) | not", variables: { $var: "nginx" } },
         },
       } as any;
-      const wrapper = mountWrapper(defaultStore, <AppRepoForm {...defaultProps} repo={repo} />);
-
+      let wrapper: any;
+      await act(async () => {
+        wrapper = mountWrapper(defaultStore, <AppRepoForm {...defaultProps} repo={repo} />);
+      });
       await waitFor(() => {
+        wrapper.update();
         expect(wrapper.find("textarea").at(0).prop("value")).toBe("nginx");
```

Comment:
Even if the tests were passing, they were throwing a message like

````
When testing, code that causes React state updates should be wrapped into act(...):
     
     act(() => {
       /* fire events that update state */
     });
     /* assert on the output */
     
This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
````

Since mounting `AppRepoForm` triggers an state update via `useEffect` in an async fetch function, it has to be wrapped into an `act`. Then, a `.update()` call is often required.

I've performed this change in the remaining test cases.

Reply:
Excellent thanks. Yeah, we'd started using `act` for async tests in other places, but only as errors were raised in tests. Thanks for doing them all here.